src/services/urlAssembly.spec.ts

```diff
@@ -60,6 +75,22 @@ describe('path params', () => {
     expect(url).toBe('/simple/ABC')
   })
 
+  test('given route with default string param provided, returns route Path with string with values interpolated', () => {
+    const [route] = createRoutes([
+      {
+        name: 'simple',
+        path: path('/simple/[simple]', { simple: withDefault(String, 'abc') }),
+        component,
+      },
+    ])
+
+    const url = assembleUrl(route, {
+      params: { simple: 'DEF' },
+    })
+
+    expect(url).toBe('/simple/DEF')
+  })
+
```

Comment:
Seems like maybe we don't need this test. The param having a default doesn't factor into this logic or types does it?